Refactor: Signup Free spec into an onboarding flow spec.

youbrokesomething opened the pull request:
#### Proposed Changes

This PR renames signup__free to heroflow__write and removes the new site sign up flow which is out of scope for onboarding.

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Ensure the following:
- [x] Calypso E2E (desktop)
- [x] Calypso E2E (mobile)
- [x] Pre-Release E2E

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #64418

youbrokesomething committed:
Refactoring the signup__free spec into an onboarding flow spec.

This commit renames signup__free to heroflow__write and removes the new site sign up flow which is out of scope for onboarding.

worldomonation commented on test/e2e/specs/onboarding/heroflow__write.ts:
```diff
@@ -0,0 +1,136 @@
+		} );
+
+		it( 'See design picker screen', async function () {
+			await startSiteFlow.validateOnDesignPickerScreen();
```
Looks like this step is failing on Pre-Release Tests with the following error:

```
waiting for selector ".generated-design-picker" to be visible
```

worldomonation commented on test/e2e/specs/onboarding/heroflow__write.ts:
```diff
@@ -0,0 +1,136 @@
+		it( 'Skip goals step', async function () {
+			startSiteFlow = new StartSiteFlow( page );
+
+			await page.waitForLoadState( 'networkidle' );
```
**suggestion**: if you are sticking with the `getCurrentStep` pattern, this `networkidle` call is unnecessary as the `getCurrentStep` method [already contains such wait](https://github.com/Automattic/wp-calypso/blob/a25293cee3a8135e3f9d604dcfc3d2b10327fd45/packages/calypso-e2e/src/lib/flows/start-site-flow.ts#L65).

worldomonation replied:
Personally though, I'm not a fan of the `getCurrentStep` structure as I feel an E2E test should take the user down a list of predefined expectations instead of allowing a way out with `if`s.

worldomonation commented on test/e2e/specs/onboarding/heroflow__write.ts:
```diff
@@ -0,0 +1,136 @@
+
+declare const browser: Browser;
+
+describe( DataHelper.createSuiteTitle( 'Onboarding heroflow: write' ), function () {
```
**suggestion::nonblocking**: could the suite name be renamed to `FTME: Write` to be in sync with [this spec file](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/specs/onboarding/ftme__sell.ts)?

In similar vein..

**suggestion::nonblocking**: could the file name be renamed to `ftme__write`, for similar reasons?

worldomonation reviewed (changes requested):
Good start and a great simplification of the existing Signup: Free spec.

I rant the spec in TeamCity [run here](https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Pre_Release/8463388?buildTab=overview&hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildTestsSection=true#%2Ftrace;%2Fscreenshots) and the spec fails at the **See design picker screen** step.

worldomonation commented:
I'm taking over the PR.

worldomonation committed:
specs/onboarding/ftme__sell.ts
- fix incorrect naming of a step.

specs/onboarding/ftme__write.ts
specs/onboarding/heroflow__write.ts
- rename spec to be in line with the other FTME spec.
- refactor the spec to start from the beginning of the site creation flow.

worldomonation committed:
packages/calypso-e2e/src/lib/pages/signup/signup-pick-plan-page.ts
- cast the blogID attribute of the returned response to number.

worldomonation committed:
packages/calypso-e2e/src/lib/pages/editor-page.ts
- add new `getLocator` method.

worldomonation committed:
packages/calypso-e2e/src/rest-api-client.ts
- remove check for blog name when deleting the site.

worldomonation committed:
test/e2e/specs/onboarding/ftme__write.ts
- check for the first post congratulatory message.

worldomonation committed:
packages/calypso-e2e/src/rest-api-client.ts
- simplify checks on the deleteSite method.

test/e2e/specs/shared/api-delete-site.ts
- restructure error message to denote that if a `null` is received that nothing took place.

worldomonation committed:
packages/calypso-e2e/src/lib/flows/start-site-flow.ts
- revert changes to selector
